src/citrine/builders/design_spaces.py

```diff
@@ -0,0 +1,96 @@
+import pandas as pd
+from itertools import product
+
+def EnumeratedProductDesignSpaceData(dof_grids:dict):
+    ''' Given a dict of lists or tuples of values for each degree of freedom,
+    create the list of candidates representing the outer product of all values
+    for each degree of freedom
+    
+    Parameters
+    ----------
+    dof_grids: dict(str:list)
+        {'degree of freedom':[list, of, values]} for each DoF
+    '''
+    design_space_tuples = list(product(*dof_grids.values()))
+    design_space_cols = list(dof_grids.keys())
+    df_ds = pd.DataFrame(data=design_space_tuples, columns=design_space_cols)
+    candidates = df_ds.to_dict('records')
+    return candidates
+
+
+def SimpleMixtureProductDesignSpaceData(formulation_grids:dict,
+                                        balance_component:str
+                                        ):
```

Comment:
```suggestion
                                        ) -> List[Mapping[str, Any]]:
```
I _think_ this is the right type.

Reply:
How about for the input (`formulation_grids`)? A function call looks like:

```
form_ds = SimpleMixtureProductDesignSpaceData({
    'balance_comp':[0.3,0.9],
    'comp1':[0,0.01,0.02,0.03,0.04,0.05],
    'comp2':[0.1,0.11,0.12,0.13,0.14,0.15],
    'comp3':[0.1,0.15,0.2,0.25,0.3],
    'comp4':[0,0.01,0.02,0.03,0.04,0.05]
},
balance_component='balance_comp')
```

Reply:
Mapping[str, Sequence], right?

Reply:
The thing after the `-> ` is the type of the return value..  How about `Mapping[str, List[float]]` for `formulation_grids`?

Reply:
Yep, note that I have updated this with a new commit